Gems/DiffuseProbeGrid/Code/Source/Components/DiffuseProbeGridSystemComponent.cpp

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Serialization/SerializeContext.h>
+#include <Atom/RPI.Public/Pass/PassSystemInterface.h>
+#include <Atom/RPI.Public/FeatureProcessorFactory.h>
+#include <Components/DiffuseProbeGridSystemComponent.h>
+#include <Components/DiffuseProbeGridComponentController.h>
+#include <Render/DiffuseProbeGridFeatureProcessor.h>
+#include <Render/DiffuseGlobalIlluminationFeatureProcessor.h>
+#include <Render/DiffuseProbeGridBlendDistancePass.h>
+#include <Render/DiffuseProbeGridBlendIrradiancePass.h>
+#include <Render/DiffuseProbeGridBorderUpdatePass.h>
+#include <Render/DiffuseProbeGridClassificationPass.h>
+#include <Render/DiffuseProbeGridDownsamplePass.h>
+#include <Render/DiffuseProbeGridPreparePass.h>
+#include <Render/DiffuseProbeGridQueryPass.h>
+#include <Render/DiffuseProbeGridRayTracingPass.h>
+#include <Render/DiffuseProbeGridRelocationPass.h>
+#include <Render/DiffuseProbeGridRenderPass.h>
+#include <Render/DiffuseProbeGridVisualizationAccelerationStructurePass.h>
+#include <Render/DiffuseProbeGridVisualizationCompositePass.h>
+#include <Render/DiffuseProbeGridVisualizationPreparePass.h>
+#include <Render/DiffuseProbeGridVisualizationRayTracingPass.h>
+
+namespace AZ
+{
+    namespace Render
+    {
+        void DiffuseProbeGridSystemComponent::Reflect(ReflectContext* context)
+        {
+            if (SerializeContext* serialize = azrtti_cast<SerializeContext*>(context))
+            {
+                serialize->Class<DiffuseProbeGridSystemComponent, Component>()
+                    ->Version(0)
+                    ;
+            }
+
+            DiffuseProbeGridFeatureProcessor::Reflect(context);
+            DiffuseGlobalIlluminationFeatureProcessor::Reflect(context);
+        }
+
+        void DiffuseProbeGridSystemComponent::GetProvidedServices(ComponentDescriptor::DependencyArrayType& provided)
+        {
+            provided.push_back(AZ_CRC_CE("DiffuseProbeGrid"));
+            provided.push_back(AZ_CRC_CE("DiffuseGlobalIllumination"));
+        }
+
+        void DiffuseProbeGridSystemComponent::GetIncompatibleServices(ComponentDescriptor::DependencyArrayType& incompatible)
+        {
+            incompatible.push_back(AZ_CRC_CE("DiffuseProbeGrid"));
+            incompatible.push_back(AZ_CRC_CE("DiffuseGlobalIllumination"));
+        }
+
+        void DiffuseProbeGridSystemComponent::GetRequiredServices([[maybe_unused]] ComponentDescriptor::DependencyArrayType& required)
+        {
+        }
```

Comment:
DDGI depends on the raytracing feature processor, which is defined in "CommonService" service in the Atom gem. I think it should be added.

Reply:
The DiffuseProbeGrid does not require ray tracing, since it can run in baked mode on non-RT platforms.  In addition, RayTracing is a bespoke engine component and not exposed as a service.

Reply:
Sure, for baked data its fine, but how does the gem (and the component) know that I can bake? Assume I have a new scene on a machine with no ray tracing capabilities with a DiffuseProbeGrid in it. What happens if I press *Bake Textures* on the component?

Reply:
The feature processor and DiffuseProbeGrid passes all query the RHI device to determine if the GPU supports ray tracing.  If it is not supported they disable all RT related processing.  This is done exclusively with the RHI device, the RayTracingFeatureProcessor is not involved.  So the DiffuseProbeGrid Gem does not depend on any RayTracing components in the engine when running on non-RT platforms.

If the user tries to Bake on a non-RT platform the Bake will not succeed, since all of the RT passes are disabled.  This may be an area where the user messaging can be improved, but that is beyond the scope of this PR, which is moving the existing code into a Gem.

Reply:
I see, thanks for the clarification.